feat(core): allow fully customizable indices settings in `snowowl.yml` configuration file

cmark opened the pull request:
Each index can receive its own custom settings object nested under `repository.index` via its index name. If not specified, index
initialization will fall back to the default values.

cmark committed:
feat(core): allow fully customizable indices settings in...

...snowowl.yml configuration file

Each index can receive its own custom settings object nested under
`repository.index` via its index name. If not specified, index
initialization will fall back to the default values.

cmark committed:
feat(core): reduce default shard size to 1 for all indices

Indices can be customized via the new snowowl.yml configuration key
`repository.index.[name].number_of_shards`

apeteri commented on core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/TerminologyRepository.java:
```diff
@@ -83,9 +83,13 @@ public String id() {
+			repositoryId, 
+			mapper, 
+			mappings, 
+			context.service(IndexSettings.class).forIndex(indexConfiguration, repositoryId)
```
Same for `Indexes.createIndexClient`, although the client name here is already parameterized.

apeteri commented on cis/com.b2international.snowowl.snomed.cis/src/com/b2international/snowowl/snomed/cis/SnomedIdentifierPlugin.java:
```diff
@@ -116,7 +117,12 @@ private void registerSnomedIdentifierService(final SnomedIdentifierConfiguration
+				SNOMED_IDS_INDEX, 
+				env.service(ObjectMapper.class), 
+				new Mappings(SctId.class), 
+				env.service(IndexSettings.class).forIndex(env.service(RepositoryConfiguration.class).getIndexConfiguration(), SNOMED_IDS_INDEX)
```
The request to get the specific index configuration could be pushed down to `Indexes.createIndex` itself, as it already knows about the index name passed in the first arg. Then you don't have to change all invocations.

cmark replied:
The problem is a reference to the `IndexConfiguration` class, which is a Snow Owl specific class with Snow Owl defaults, while the index layer is treated like a library and unfortunately it does not have access to that class.

cmark committed:
chore(test): add example configuration to test custom index settings

nagyo reviewed (approved):
LGTM!

cmark committed:
chore(index): make sure we stored the number_of_shards and replicas...

...in the EsIndexAdmin configuration settings map and use that to update
the underlying ES index.

cmark committed:
chore(docker): add example configuration of snomed index...

...number_of_shards

nagyo reviewed (approved):
👍🏻
